.travis.yml

```diff
@@ -7,6 +7,12 @@ branches:
     - master
     - develop
 
+before_script:
```

Comment:
please move this into make to ensure it also ends up in docker and on the machine of anyone using make without docker

Reply:
This system library install is specific to Ubuntu Trusty. I feel like the Makefile should be platform agnostic. Most newer systems will have a system package that is viable for zmq (you can probably just `apt-get install libzmq3-dev`) It's just the old travis environment that needs the special install.

I'm up for suggestions on how to do this. I could potentially add something to the Makefile that tries to do a pkg-config for libzmq, and throws an error if it doesn't find it on the system... but it probably wouldn't work on Windows..

Reply:
@neonphog the makefile only supports ubuntu right now so don't worry about windows, at least not for this merge

Reply:
@thedavidmeister - Ok, moved into Makefile, checks pkg-config so doesn't try to install if the lib exists already, mostly it will just do a system `sudo apt-get install libzmq3-dev`, or if `apt-get` doesn't exists it'll give you a warning that the build probably won't work but proceed. Two things to note:

- if the `TRAVIS` environment variable exists, it will use the custom updated version package
- requires `sudo` on all the travis tasks now